rest/context.go

```diff
@@ -72,6 +72,9 @@ type RequestContext interface {
 	// string is returned with the error set.
 	NextURL() (string, error)
 
+	// BuildURL builds the URL for the given route name and resource ID.
+	BuildURL(routeName, resourceID string) (string, error)
```

Comment:
Instead of passing a route name, could we just have constants like `Get`, `GetList`, `Create`, etc.? This would make sense to the user I think, then we could handle the internal route name without leaking it.